STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/az_ulib_dm.c

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#include "az_ulib_result.h"
+#include "az_ulib_dm_api.h"
+#include "_az_ulib_dm_interface.h"
+#include "azure/az_core.h"
+#include "cipher_v1i1.h"
+#include "cipher_v2i1.h"
+#include "sprinkler_v1i1.h"
+#include <azure/core/internal/az_precondition_internal.h>
+#include <string.h>
+
+/*
+ * DM is a singleton component, and shall be initialized only once.
+ *
+ * Make it volatile to avoid any compilation optimization.
+ */
+static az_ulib_dm* volatile _az_dm_cb = NULL;
```

Comment:
what's the difference of doing this vs "static volatile az_ulib_dm* _az_dm_cb"?

Reply:
None. but now that you mentioned, I will change to `static volatile az_ulib_dm* _az_dm_cb`.

Reply:
`static volatile az_ulib_dm* _az_dm_cb` means that the pointer `_az_dm_cb` is volatile.
`static az_ulib_dm* volatile _az_dm_cb` means that the content pointed by `_az_dm_cb` is volatile.